### ISSUE TITLE
CouchDB _users session user doc is not deleted on logout

### ISSUE THREAD

**Issue by clariontools** [colinskow/superlogin#186]
In my configuration with CouchDB 2.0, I noticed that the user "session" in the CouchDB _users db is not deleted on logout, although the user in the security document of the per user db removed.  I looked at the code and I found that in the couchdb.js this.removeKeys function the call to couchAuthDB.allDocs returns an error not_found on each session key passed to it.

I then tried to replicate the POST with {"keys": "org.couchdb.user:key-i-am-looking-for"}, using Postman or CURL and noticed that it also returns the not_found error.  I check the docs and it seems to be formatted properly but for some reason (maybe a special case for _users db) it does not find the keys.  SO I tried a single key using GET with key="org.couchdb.user:key-i-am-looking-for" and it works as expected.  Also tried, GET with keys=["org.couchdb.user:key-i-am-looking-for"] and that fails just like the POST does with each key returning a "not_found" error.

This explains all the mess I was documenting in #179 , and it would be nice to hear some confirmation of this issue of _users session docs not being deleted.  Does anyone else have this problem?  Since it appears to be an issue with my CouchDB 2.0 configuration, maybe if you don't see this problem you could confirm here what version of CouchDB you are using?

EDIT: Just tested on one of the user dbs by doing _all_docs POST with keys and it works as expected.  This points to some difference in the _users db that is causing _all_docs keys (with POST or GET) to fail where a single key with GET succeeds.

**Comment by clariontools** [comment]
I posted an issue over on the couchdb github repo after doing more testing to validate this issue.  I also verified that the actual delete via bulkDocs does work as is in the couchdb.js source, so that part is good.  A workaround would be to make multiple call to all_docs using GET for each key to validate, but that seems silly.  The other option is to not validate the keys at all and simply call bulkDocs to delete, but this is not possible without having the current _rev value, so for now it is individual GET calls for a workaround.

**Comment by clariontools** [comment]
While I am waiting for a response from the couchdb github repo issue I posted, I did implement the workaround using GET with a single key, and it works exactly as expected for a superlogin logout of a single session, but would need do some more work when used with a multiple session logout call such as, logoutAll etc...  Now that I have a solution for one of the problems I identified in #179, I think I will try to add the per user db 'user:<user_id> as the member in order to eliminate the case of public access on the per user db when no user sessions exist.

**Comment by clariontools** [comment]
Just heard back from the CouchDB github repo and they say the problem is not reproducible in CouchDB 2.1, so it could be it was fixed for the first time or was a bug in CouchDB 2.0.  I will give 2.1 a try and report back here with the results for future reference.

**Comment by clariontools** [comment]
Confirmed this is not an issue with superlogin, it is due to a bug in CouchDB 2.0 so I would advise not using CouchDB 2.0 with superlogin.  I have upgraded and tested with CouchDB 2.1 and it resolves the issue.  There is still the security issue noted in #179 , but I will also test that, maybe it too has been resolved by CouchDB 2.1.  Here is my closed issue from the couchdb github repo: https://github.com/apache/couchdb/issues/878
